lib/SIL/IR/SILFunction.cpp

```diff
@@ -553,6 +553,32 @@ bool SILFunction::isNoReturnFunction(TypeExpansionContext context) const {
       .isNoReturnFunction(getModule(), context);
 }
 
+bool SILFunction::hasNonUniqueDefinition() const {
+  // Non-uniqueness is a property of the Embedded linkage model.
+  if (!getASTContext().LangOpts.hasFeature(Feature::Embedded))
+    return false;
+
+  /// The entrypoint always has a unique definition.
+  if (getDeclRef().kind == SILDeclRef::Kind::EntryPoint)
+    return false;
+
+  /// A @_used symbol has a unique definition.
```

Comment:
This seems like a conflation, no?

Reply:
Perhaps. Without this, we end up with a problem where an Embedded Swift library that defines a main entry point (e.g., via `@_used @cdecl("main")`) would not link, the `main` isn't used in an importing module. There may be a better way to do this, e.g., by forcing all imported `@_used` symbols to be referenced from `llvm.used`.

Reply:
FWIW, [this is the test that fails](https://github.com/swiftlang/swift/blob/main/test/embedded/modules-used2.swift) when I don't handle "used" like this.

Reply:
Found a better solution!